core/trino-main/src/test/java/io/trino/operator/scalar/TestArrayExceptFunction.java

```diff
@@ -66,4 +66,10 @@ public void testDuplicates()
         assertFunction("array_except(ARRAY[VARCHAR 'x', 'x', 'y', 'z'], ARRAY['x', 'y', 'x'])", new ArrayType(VARCHAR), ImmutableList.of("z"));
         assertFunction("array_except(ARRAY[true, false, null, true, false, null], ARRAY[true, true, true])", new ArrayType(BOOLEAN), asList(false, null));
     }
+
+    @Test
+    public void testNonDistinctEqualValues()
+    {
+        assertFunction("array_except(ARRAY[NaN()], ARRAY[NaN()])", new ArrayType(DOUBLE), ImmutableList.of());
```

Comment:
Mix with some normal numbers, so we do not end up with empty result which is kinda a corner case.